lib/core/test/unit/Cardano/Pool/MetricsSpec.hs

```diff
@@ -47,6 +49,24 @@ spec = do
                 let count = Map.foldl (\r l -> r + (length l)) 0 . activity
                 count s' === (count s + 1)
 
+        describe "combineMetrics" $ do
+            it "pools with no entry for productions are included" $
+                property $ \stakeDistr -> do
+                    combineMetrics stakeDistr Map.empty
+                    `shouldBe`
+                    (Right $ Map.map (, 0) stakeDistr)
+
+            it "it fails if a block-producer is not in the stake distr" $ do
+                property $ \stakeDistr poolProd ->
+                    let
+                        aPoolWithoutStakeProduced =
+                            not . Map.null $ Map.difference poolProd stakeDistr
+                    in classify aPoolWithoutStakeProduced
+                        "A pool without stake produced" $
+                        combineMetrics stakeDistr poolProd
+                        `shouldSatisfy` \case
+                            Left _ -> aPoolWithoutStakeProduced
+                            Right _x -> True
```

Comment:
These two properties, I think, are the _particularly_ interesting cases… although maybe a bit thin…

Reply:
I think it gives a nice coverage and a good sense of what we expect.
What about some coverage check for the classification to make sure that we are actually testing both cases?